src/super_gradients/training/samples/segmentation_sample.py

```diff
@@ -0,0 +1,31 @@
+import dataclasses
+from typing import Union
+
+import numpy as np
+import torch
+from PIL.Image import Image
+
+__all__ = ["SegmentationSample"]
+
+
+@dataclasses.dataclass
+class SegmentationSample:
+    """
+    A data class describing a single object detection sample that comes from a dataset.
+    It contains both input image and target information to train an object detection model.
+
+    :param image:              Associated image with a sample. Can be in [H,W,C] or [C,H,W] format
+    :param bboxes_xywh:        Numpy array of [N,4] shape with bounding box of each instance (XYWH)
+    :param labels:             Numpy array of [N] shape with class label for each instance
+    :param is_crowd:           (Optional) Numpy array of [N] shape with is_crowd flag for each instance
+    :param additional_samples: (Optional) List of additional samples for the same image.
+    """
+
+    __slots__ = ["image", "mask"]
+
+    image: Union[np.ndarray, torch.Tensor]
+    mask: Union[np.ndarray, torch.Tensor]
+
+    def __init__(self, image: Union[np.ndarray, torch.Tensor, Image], mask: Union[np.ndarray, torch.Tensor, Image]):
+        self.image = image
```

Comment:
If we want to allow input to be PIL image, we can convert it here to np.ndarray in __init__ since both Albu works only with np.ndarray and conversion to tensor also uses numpy and not PIL image

Reply:
The reason for the support for PIL image is because this is what all of our segmentation transforms expect.
What we should aim for IMO is to support PIL images in segmentation sample and gradually reproduce our recipes with albumentations transforms, then at that point deprecate and eventually remove them.